### PR TITLE
chore: add additional test cases to reset password component service

### PR DESCRIPTION
**Pull request by kpawelczak**
closes: [CXSPA-8984](https://jira.tools.sap/browse/CXSPA-8984)

### CODE REVIEW

File: feature-libs/user/profile/components/reset-password/reset-password-component.service.spec.ts [hunk 1]
```diff
@@ -82,166 +85,251 @@ describe('ResetPasswordComponentService', () => {
...
+              throwError(() => error)
+            );
+            service.resetPassword(resetToken);
+            expect(globalMessageService.add).toHaveBeenCalledWith(
+              { raw: 'error message' },
+              GlobalMessageType.MSG_TYPE_ERROR
+            );
+          });
+
+          it('should not show error message', () => {
```
Comment (rmch91) [inline feature-libs/user/profile/components/reset-password/reset-password-component.service.spec.ts]: duplicated name `should not show error message`

**Review by rmch91 — CHANGES_REQUESTED** [review]

File: feature-libs/user/profile/components/reset-password/reset-password-component.service.spec.ts (hunk 1 above)
Comment (rmch91) [reply feature-libs/user/profile/components/reset-password/reset-password-component.service.spec.ts]: or duplicated test case, not sure

Comment (kpawelczak) [reply feature-libs/user/profile/components/reset-password/reset-password-component.service.spec.ts]: fixed

**Review by rmch91 — DISMISSED** [review]
LGTM

**Review by rmch91 — APPROVED** [review]

**Pull request merged**